htdocs/admin/content/alliance/edit.inc.php

```diff
@@ -1,638 +1,534 @@
 <?PHP
 
-		if (isset($_GET['alliance_id']))
-			$id = $_GET['alliance_id'];
-		if (isset($_GET['id']))
-			$id = $_GET['id'];
-
-
-		if (isset($_POST['info_save']) && $_POST['info_save']!="")
-		{
-			//  Bild löschen wenn nötig
-			$img_sql="";
-			if (isset($_POST['alliance_img_del']))
-			{
-				$res = dbquery("SELECT alliance_img FROM alliances WHERE alliance_id=".$id.";");
-				if (mysql_num_rows($res)>0)
-				{
-					$arr=mysql_fetch_array($res);
-					if (file_exists('../'.ALLIANCE_IMG_DIR."/".$arr['alliance_img']))
-					{
-						unlink('../'.ALLIANCE_IMG_DIR."/".$arr['alliance_img']);
-					}
-					$img_sql=",alliance_img=''";
-				}
+use EtoA\Alliance\AllianceBuildingRepository;
+use EtoA\Alliance\AllianceRepository;
+use EtoA\Alliance\AllianceTechnologyRepository;
+
+/** @var AllianceRepository */
+$repository = $app['etoa.alliance.repository'];
+
+/** @var AllianceBuildingRepository */
+$buildingRepository = $app['etoa.alliance.building.repository'];
+
+/** @var AllianceTechnologyRepository */
+$technologyRepository = $app['etoa.alliance.technology.repository'];
+
+if (isset($_GET['alliance_id'])) {
+	$id = $_GET['alliance_id'];
```

Comment:
Instead of relying on GLOBALS there is also a symfony request object around (not sure whether this is initialized in every entry file already. Would for instance allow us to use:

```
$id = $request->query->getInt('alliance_id');
```

Enforcing that the id is an integer

Reply:
Done in ed1b97f